### DISCUSSION TITLE
Need Help with Setup

### DISCUSSION THREAD

**Discussion by Gymbow2007** [maca88/SmartBikeLights#55] [General]
I have an Edge 840 with an Ion 200 RT Light.  I’m confused as to how to use the filters to accomplish the following.  
1. When my speed is >= 3 mph, turn ON light.  Alternately, I’m ok with using the 840 Timer start to turn the light ON.
2. When my speed drops below 3 mph, turn the light OFF

Appreciate any suggestions on how to configure this.

**Comment by Gymbow2007** [comment]
I should clarify the above.  After I stop (ie speed < 3 mph) and the light turns OFF, when I continue my ride (ie speed >= 3 mph), I want the light to turn back ON.

**Comment by maca88** [comment]
Hi,

Here is the configuration for what you want to achieve:

1,1!:1H]0#4587520,196641::1#2,2!ON:1:2:0:0C}1.34112!:1:0:0:0D=1###7,6:Ion 200 RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62##0::#0:0#123!:123!#0#0#B4062#2##1#0

I've added a global filter, so that the light will turn on only when the activity is running. You can remove it if you want, it is not required.
Other than that the configuration is very simple, the default mode is Off and the light mode will switch to Medium only when the current speed is greater or equal 3 mph.

**Comment by Gymbow2007** [comment]
Thank You for your help, it’s greatly appreciated.  If possible, could you take a snapshot of these settings in the Light Configuration Tool as that will give me a much better understanding of how to use that tool if I want to make any changes?  Again, Thank You for your help & for creating such a useful app.

**Reply by maca88** [reply]
> if I want to make any changes?

You can copy the above configuration, paste it in the "Existing configuration" input field of the [Lights Configurator](https://maca88.github.io/SmartBikeLights/) and click on the "LOAD" button to load it.

**Comment by Gymbow2007** [comment]
Regarding my request for a snapshot of the Light Configuration Tool, never mind.  I realized I can load your file into the tool to see the details.  Thanks again for everything.

**Comment by Gymbow2007** [comment]
﻿Hi, Thanks for that info.  I made a couple of minor tweaks, but nothing that would change what you provided.  Here is what happens with this configuration. - When I press the Start Activity Timer, the light turns ON- When I pause the ride (5-10 minute stop at the park), the light remains ON- The light will turn OFF when I Stop Activity Timer at the end of my ride1,1!:1H]0#4587520,196641::1#2,2!ON:1:63:5:1C}1.34112!:1:63:0:0D=1###7,6:Ion 200 RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62##0::#0:0#123!:123!#1#0#B4062#2##1#1What I want to have happen when I pause the ride at the Park (or drop my speed below 3 mph) is to have the light turn OFF.  Then when I resume riding again, I want the light to turn ON.How do I accomplish this?Thanks,JimOn Dec 11, 2023, at 3:32 PM, maca88 ***@***.***> wrote:﻿

if I want to make any changes?

You can copy the above configuration, paste it in the "Existing configuration" input field of the Lights Configurator and click on the "LOAD" button to load it.

**Reply by maca88** [reply]
>  I made a couple of minor tweaks, but nothing that would change what you provided

You changed the "Default mode", which does change the behavior on how the light will behave. The default mode in your case will be used when the activity is running and the speed is lower than 3 mph, which means that the light will always be on after starting the activity. If you want that the light is turned off when the speed is lower than 3 mph then you need to set the "Default mode" to "Off".

> What I want to have happen when I pause the ride at the Park (or drop my speed below 3 mph) is to have the light turn OFF

If you want to turn off the light even after stopping the activity regardless the current speed, you can do that by modifying the global filter to "Timer state > Recording stopped" instead of "Timer state > Not recording".

Here is the updated configuration:

1,1!:1H]1#4587520,196641::1#2,2!ON:1:63:5:1C}1.34112!:1:0:0:0D=1###7,6:Ion 200 RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62##0::#0:0#123!:123!#1#0#B4062#2##1#1

**Comment by Gymbow2007** [comment]
Ah ha!  I get it now & understand how the tool works.  Awesome!ThanksOn Dec 12, 2023, at 12:19 PM, maca88 ***@***.***> wrote:1,1!:1H]1#4587520,196641::1#2,2!ON:1:63:5:1C}1.34112!:1:0:0:0D=1###7,6:Ion 200 RT:0:16777215!2,:-1,Off:0!1,High:1!1,Medium:2!1,Low:5!1,Day Flash:63!1,Night Flash:62##0::#0:0#123!:123!#1#0#B4062#2##1#1
